python/packages/ai/teams/ai/planner/openai_planner.py

```diff
@@ -144,25 +153,71 @@ async def _complete_prompt(
-        result = await self._complete_text(prompt_template)
+        if is_chat_completion:
+            user_message = turn_state.temp.value.input
+            result = await self._complete_chat(
+                prompt_template, turn_state, user_message, history_options
+            )
+        else:
+            result = await self._complete_text(prompt_template)
 
         duration = datetime.now() - start_time
         # TODO: investigate how to get prompt/completion tokens
         self._log_request(f"\n{log_prefix} SUCCEEDED: duration={duration} response={result}")
 
         return result
 
+    async def _complete_chat(
+        self,
+        prompt_template: PromptTemplate,
+        state: TurnState,
+        user_message: str,
+        options: AIHistoryOptions,
+    ):
+        prompt_template_config = generate_sk_prompt_template_config(prompt_template)
+        request_settings = ChatRequestSettings.from_completion_config(
+            prompt_template_config.completion
+        )
+
+        chat_completion_client = self._sk.get_ai_service(ChatCompletionClientBase)(self._sk)
+
+        chat_history: List[Tuple[str, str]] = []
+        if self._options.use_system_message:
+            chat_history.append((SYSTEM_ROLE_NAME, prompt_template.text))
+        else:
+            chat_history.append((USER_ROLE_NAME, prompt_template.text))
+
+        # Populate conversation history
+        if options.track_history:
+            user_prefix = options.user_prefix
+            assistant_prefix = options.assistant_prefix
+            history: List[str] = state_to_history_array(state, options.max_tokens)
+
+            for line in history:
+                if line.lower().startswith(user_prefix.lower()):
+                    chat_history.append((USER_ROLE_NAME, line[len(user_prefix) :].strip()))
+                elif line.lower().startswith(assistant_prefix.lower()):
+                    chat_history.append(
+                        (ASSISTANT_ROLE_NAME, line[len(assistant_prefix) :].strip())
+                    )
+
+        if user_message:
+            chat_history.append((USER_ROLE_NAME, user_message))
+
+        result = await chat_completion_client.complete_chat_async(chat_history, request_settings)
+        if isinstance(result, str):
+            return result
+
+        return "\n".join(result)
+
     async def _complete_text(self, prompt_template: PromptTemplate):
         prompt_template_config = generate_sk_prompt_template_config(prompt_template)
         request_settings = CompleteRequestSettings.from_completion_config(
             prompt_template_config.completion
         )
 
-        text_completion_client = self._sk.get_ai_service(TextCompletionClientBase)(self)
-        result = await text_completion_client.complete_async(
-            prompt_template.text, request_settings, NullLogger()
-        )
+        text_completion_client = self._sk.get_ai_service(TextCompletionClientBase)(self._sk)
```

Comment:
like above, this code doesn't work when run

Reply:
See my previous comment about this. Resolving this to reduce concurrent discussions on same topic. We can discuss further in the other thread.